jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceUnitMetadata.java

```diff
@@ -90,4 +90,12 @@ public interface PersistenceUnitMetadata extends PersistenceUnitInfo {
 
     List<ClassTransformer> getTransformers();
 
+    Object getLockObject();
+
+    /**
+      * @return true if a DelegatingClassTransformer was already setup to transform application classes for this persistence unit.
+     */
+    boolean wasDelegatingTransformerAdded();
+
+    void setDelegatingTransformerAdded();
```

Comment:
IMO this should simply be a CAS. Something like:
```java
boolean compareAndSetDelegatingTransformerAdded(boolean expected, boolean value);
```

Reply:
Switched to a compare and set, thanks!  And just pushed the change.